Develop

Ikumi-Kobayashi opened the pull request.

Commit:
add test code

Ikumi-Kobayashi committed:
重複の書き方を一行にしました。

Ikumi-Kobayashi committed:
重複の書き方を一行にしました。
３つあった　it'カードが５個じゃないとき' をもっと具体的にしました。

Ikumi-Kobayashi committed:
重複の書き方を一行にしました。
３つあった　it'カードが５個じゃないとき' をもっと具体的にしました。
controllerのテストも書いたのですが、うまくいきません。レビューお願いします。

Ikumi-Kobayashi committed:
重複の書き方を一行にしました。
３つあった　it'カードが５個じゃないとき' をもっと具体的にしました。
controllerのテストも書いたのですが、うまくいきません。（poker_controller_spec.rbの11~19行目のメモにしてあるところ）レビューお願いします。

Ikumi-Kobayashi committed:
重複の書き方を一行にしました。
３つあった　it'カードが５個じゃないとき' をもっと具体的にしました。
controllerのテストも書いたのですが、うまくいきません。（poker_controller_spec.rbの11~19行目のメモにしてあるところ）レビューお願いします。
include PokerCheckServiceをpoker_check_service_spec.rbでしても、PokerCheckService.new("")で記載するとエラーになってしまったので、そのままの書き方で置いてあります。

Ikumi-Kobayashi committed:
test modified

Ikumi-Kobayashi committed:
make API code

Ikumi-Kobayashi committed:
apiのファイル名をcheck_api.rb、root_api.rbに変更しました。

Ikumi-Kobayashi committed:
apiのファイル名をcheck_api.rb、root_api.rbに変更しました。
新たに作ったサービスを削除し、既存のサービスにstrongアクションを追加する形にしました。

Ikumi-Kobayashi committed:
apiのファイル名をcheck_api.rb、root_api.rbに変更しました。
新たに作ったサービスを削除し、既存のサービスにstrongアクションを追加する形にしました。
api_resultの変数定義の場所を変更しました。

Ikumi-Kobayashi committed:
apiのファイル名をcheck_api.rb、root_api.rbに変更しました。
新たに作ったサービスを削除し、既存のサービスにstrongアクションを追加する形にしました。
api_resultの変数定義の場所を変更しました。
zipメソッドでまとめて処理していたところを、仏様な処理ごとに分割しました。

Ikumi-Kobayashi committed:
apiのファイル名をcheck_api.rb、root_api.rbに変更しました。
新たに作ったサービスを削除し、既存のサービスにstrongアクションを追加する形にしました。
api_resultの変数定義の場所を変更しました。
zipメソッドでまとめて処理していたところを、必要な処理ごとに分割しました。
entityを用いて処理を書き直しました。

Ikumi-Kobayashi committed:
modified API code

Ikumi-Kobayashi committed:
modified API code

Ikumi-Kobayashi committed:
課題２および課題２のテストコードを作成しました。

Ikumi-Kobayashi committed:
Merge pull request #2 from Ikumi-Kobayashi/feature-poker-webapp

add test code

shnzw124 commented on app/apis/api/base/api.rb:
```diff
@@ -0,0 +1,13 @@
+    #404の時（URLの形がおかしい　”不正なURLです。”）
+    route :any, '*path' do
+      error!({error: "404 Not Found：不正なURLです。"}, 404)
+    end
```
エラーハンドリングの関してですが
Internal Server Errorの場合にも例外を補足してエラーメッセージを返却するように実装してください。

shnzw124 commented on app/apis/api/v1/check_api.rb:
```diff
@@ -0,0 +1,69 @@
+module V1
+  class Check_Api < Grape::API
```
Rubyではクラス名はキャメルケース（ex. CheckApi）、ファイル名はスネークケース（ex. check_api.rb）で書く
というルールがあります。
クラス名はキャメルケースで書きましょう。

shnzw124 commented on app/apis/api/v1/check_api.rb:
```diff
@@ -0,0 +1,69 @@
+    resources :check_api do
+      post '/' do
+        cards = params[:cards]
+        error_message_array = []
```
配列の変数名ですが、○○_arrayとするよりも
変数の中に複数のデータが含まれているということで○○sと複数形にしたほうがよいです。

shnzw124 commented on app/apis/api/v1/check_api.rb:
```diff
@@ -0,0 +1,69 @@
+        card_array = []
+        cards.each do |card|
+          check_service = PokerCheckService.new(card)
+          @error_message = check_service.error
```
PokerCheckServiceのバリデーションメソッドですが、
errorよりもvalidやvalid?など動詞的に表現したほうがよいかとおもいます。
errorとすると、そのインスタンスのフィールドにアクセスしてるような印象を受けるので

shnzw124 commented on app/apis/api/v1/check_api.rb:
```diff
@@ -0,0 +1,69 @@
+          card_array.push(card)
+          if @error_message.present?
+            error_message_array.push(@error_message)
+            result_array.push(nil )
```
nilの後のスペース消しておきましょう

shnzw124 commented on app/apis/api/v1/check_api.rb:
```diff
@@ -0,0 +1,69 @@
+          if @error_message.present?
+            error_message_array.push(@error_message)
+            result_array.push(nil )
+            strong_number_array.push(10 )
```
こちらも同様にスペースを消しておきましょう
こうしたスペースがバグの原因になるケースもあるので

shnzw124 commented on app/apis/api/v1/check_api.rb:
```diff
@@ -0,0 +1,69 @@
+          else
+            error_message_array.push(nil)
+            judge_service = PokerCheckService.new(card)
+            @result = judge_service.result
```
PokerCheckServiceの役判定用メソッドですが、
resultよりもcheckやjudgeなどとしたほうがよいかとおもいます。
上でも書きましたがresultとすると、そのインスタンスのフィールドにアクセスしてるような印象を受けます

shnzw124 commented on app/services/poker_check_service.rb:
```diff
@@ -46,8 +46,9 @@ def error
         #カードの重複
-       else  @card.count - @card.uniq.count > 0
-         @error_message = "カードが重複しています。"
+          if @card.count - @card.uniq.count > 0
```
意味的には変わらないですが、@card.count - @card.uniq.count > 0 とするよりも
@card.uniq.count != @card.count としたほうが理解しやすいかとおもいます。

基本的に、コードを読む順序も一般的な文章を読む順序と変わらないとおもうので
左から右に、上から下に読んでいったときに一番理解しやすいように書くのがよいとおもいます。

shnzw124 commented on app/services/poker_check_service.rb:
```diff
@@ -79,7 +80,7 @@ def result
       elsif suits[0] == suits[1] && suits[1] == suits[2] && suits[2] == suits[3] && suits[3] == suits[4]
```
suits.uniq.size == 1 などとしても同様に表現できます

shnzw124 commented on app/services/poker_check_service.rb:
```diff
@@ -119,6 +120,29 @@ def result
     end
 
+    def strong
+      if  @result == "ストレートフラッシュ"
```
条件分岐が3つ以上になる場合はcase文の導入を検討しましょう

shnzw124 commented on lib/api/api.rb:
```diff
@@ -0,0 +1,6 @@
+class API < Grape::API
```
こちらのファイルは不要なので消しておきましょう

shnzw124 commented on lib/api/document_api.rb:
```diff
@@ -0,0 +1,63 @@
+# class Document_API < Grape::API
```
こちらも消しておきましょう

Ikumi-Kobayashi committed:
internal server errorを表記しました。

Ikumi-Kobayashi committed:
internal server errorを表記しました。
クラス名をキャメルケースに変更しました。

Ikumi-Kobayashi committed:
internal server errorを表記しました。
クラス名をキャメルケースに変更しました。
変数名を○○_arrayの形から、○○sの形に変更しました。

Ikumi-Kobayashi committed:
internal server errorを表記しました。
クラス名をキャメルケースに変更しました。
変数名を○○_arrayの形から、○○sの形に変更しました。
サービスのerrorメソッドの名前をvalid?メソッドに変更しました。

Ikumi-Kobayashi committed:
internal server errorを表記しました。
クラス名をキャメルケースに変更しました。
変数名を○○_arrayの形から、○○sの形に変更しました。
サービスのerrorメソッドの名前をvalid?メソッドに変更しました。
nil後のスペースを消しました。

Ikumi-Kobayashi committed:
internal server errorを表記しました。
クラス名をキャメルケースに変更しました。
変数名を○○_arrayの形から、○○sの形に変更しました。
サービスのerrorメソッドの名前をvalid?メソッドに,resultメソッドの名前をjudgeメソッドに変更しました。
nil後のスペースを消しました。

Ikumi-Kobayashi committed:
internal server errorを表記しました。
クラス名をキャメルケースに変更しました。
変数名を○○_arrayの形から、○○sの形に変更しました。
サービスのerrorメソッドの名前をvalid?メソッドに,resultメソッドの名前をjudgeメソッドに変更しました。
nil後のスペースを消しました。
フラッシュの書き方を変更しました。

Ikumi-Kobayashi committed:
internal server errorを表記しました。
クラス名をキャメルケースに変更しました。
変数名を○○_arrayの形から、○○sの形に変更しました。
サービスのerrorメソッドの名前をvalid?メソッドに,resultメソッドの名前をjudgeメソッドに変更しました。
nil後のスペースを消しました。
フラッシュの書き方を変更しました。
case文をstrongアクションに導入しました。

Ikumi-Kobayashi committed:
internal server errorを表記しました。
クラス名をキャメルケースに変更しました。
変数名を○○_arrayの形から、○○sの形に変更しました。
サービスのerrorメソッドの名前をvalid?メソッドに,resultメソッドの名前をjudgeメソッドに変更しました。
nil後のスペースを消しました。
フラッシュの書き方を変更しました。
case文をstrongアクションに導入しました。
不要なファイルを削除しました。

Ikumi-Kobayashi committed:
レビューを受けて修正しました

Ikumi-Kobayashi committed:
Merge pull request #5 from Ikumi-Kobayashi/feature-poker-webapp

あ
